test/unit/remote-config/remote-config.spec.ts

```diff
@@ -394,6 +548,51 @@ describe('RemoteConfig', () => {
           'message', `Remote Config conditions must be an array`);
     });
 
+    it('should resolve with parameters:{} when no parameters present in the response', () => {
+      const response = deepCopy({ conditions: [], parameterGroups: {}, etag: '0-1010-2' });
+      const stub = sinon
+        .stub(RemoteConfigApiClient.prototype, 'publishTemplate')
+        .resolves(response);
+      stubs.push(stub);
+      return remoteConfig.publishTemplate(REMOTE_CONFIG_TEMPLATE)
+        .then((template) => {
+          expect(template.conditions).deep.equals([]);
+          // if parameters are not present in the response, we set it to an empty object.
+          expect(template.parameters).deep.equals({});
+          expect(template.parameterGroups).deep.equals({});
+        });
+    });
+
+    it('should resolve with parameterGroups:{} when no parameter groups present in the response', () => {
+      const response = deepCopy({ conditions: [], parameters: {}, etag: '0-1010-2' });
+      const stub = sinon
+        .stub(RemoteConfigApiClient.prototype, 'publishTemplate')
+        .resolves(response);
+      stubs.push(stub);
+      return remoteConfig.publishTemplate(REMOTE_CONFIG_TEMPLATE)
+        .then((template) => {
+          expect(template.conditions).deep.equals([]);
+          expect(template.parameters).deep.equals({});
+          // if parameterGroups are not present in the response, we set it to an empty object.
+          expect(template.parameterGroups).deep.equals({});
+        });
+    });
+
+    it('should resolve with conditions:[] when no conditions present in the response', () => {
```

Comment:
There's a lot of redundancy in these tests. As a future improvement, think about ways to clean it up a bit.